packages/salesforcedx-vscode-apex/src/views/testOutlineProvider.ts

```diff
@@ -111,6 +113,22 @@ export class ApexTestOutlineProvider
     this.onDidChangeTestData.fire();
   }
 
+  public async onResultFileCreate(
+    apexTestPath: string,
+    testResultFile: string
+  ) {
+    await this.refresh();
+    const testRunIdFile = path.join(apexTestPath, 'test-run-id.txt');
+    const testRunId = fs.readFileSync(testRunIdFile);
+    const testResultFilePath = path.join(
+      apexTestPath,
+      'test-result-' + testRunId + '.json'
+    );
+    if (testResultFile === testResultFilePath) {
+      this.readJSONFile(testResultFile);
```

Comment:
Currently, I'm creating a watcher on the creation of the new json test result file and verifying that the file path returned from that matches the test id in the `test-result-id.txt` file. I haven't handled the scenario where the two don't match yet; `readJSONFile()` doesn't get called and the tests don't get updated right now.